lib/iris/tests/integration/test_netcdf__loadsaveattrs.py

```diff
@@ -1013,4 +1223,84 @@ def test_12_localstyle__multiple_different(self, local_attr):
                 "v a l u e - A",
                 "v a l u e - B",
             ]
-        assert results == expected_results
+        elif local_attr == "STASH":
+            # A special case : the stored name is different
+            self.attrname = "um_stash_source"
+        self.check_save_results(expected_results)
+
+    #
+    # Test handling of newstyle independent global+local cube attributes.
+    #
+    def test_globallocal_clashing(self, do_split):
+        # A cube has clashing local + global attrs.
+        original_values = ["valueA", "valueB"]
+        self.run_save_testcase("userattr", original_values, expect_warnings=[])
+        expected_result = original_values.copy()
+        if not do_split:
+            # in legacy mode, "promote" = lose the local one
+            expected_result[0] = expected_result[1]
+            expected_result[1] = None
+        self.check_save_results(expected_result)
+
+    def test_globallocal_oneeach_same(self, do_split):
+        # One cube with global attr, another with identical local one.
+        inputs = [[None, "value"], ["value", None]]
+        if do_split:
+            expected = [None, "value", "value"]
+            warning = (
+                "Saving the cube global attributes \\['userattr'\\] as local"
+            )
+        else:
+            # N.B. legacy code sees only two equal values (and promotes).
+            expected = ["value", None, None]
+            warning = None
+
+        self.run_save_testcase(
+            "userattr", values=inputs, expect_warnings=warning
+        )
+        self.check_save_results(expected)
+
+    def test_globallocal_oneeach_different(self, do_split):
+        # One cube with global attr, another with a *different* local one.
+        inputs = [[None, "valueA"], ["valueB", None]]
+        if do_split:
+            warning = (
+                "Saving the cube global attributes \\['userattr'\\] as local"
+            )
+        else:
+            # N.B. legacy code does not warn of global-to-local "demotion".
+            warning = None
+        self.run_save_testcase("userattr", inputs, expect_warnings=warning)
+        self.check_save_results([None, "valueA", "valueB"])
+
+    def test_globallocal_one_other_clashingglobals(self, do_split):
+        # Two cubes with both, second cube has a clashing global attribute.
+        inputs = [["valueA", "valueB"], ["valueXXX", "valueB"]]
+        if do_split:
+            expected = [None, "valueB", "valueB"]
+            expected_warnings = [
+                "Saving.* global attributes.* as local",
+                'attributes.* of cube "v1" have been lost',
+                'attributes.* of cube "v2" have been lost',
```

Comment:
```suggestion
                'attributes.* of cube "v1" were not saved',
                'attributes.* of cube "v2" were not saved',
```

Since 80a4039647e469c2d055be215910202820cf6476

Reply:
As above https://github.com/SciTools/iris/commit/fb343aeb96ebe95a802e6e926918039526fc39db